numpy/testing/tests/test_utils.py

```diff
@@ -152,7 +152,8 @@ def test_recarrays(self):
 
         self._test_equal(a, b)
 
-        c = np.empty(2, [('floupipi', float), ('floupa', float)])
+        c = np.empty(2, [('floupipi', float),
+                         ('floupi', float), ('floupa', float)])
         c['floupipi'] = a['floupi'].copy()
         c['floupa'] = a['floupa'].copy()
```

Comment:
This change was needed because now the comparison 3 lines below here falls into the elementwise comparison case, since b and c were now castable after thie PR. So here I added an extra field to make sure b and c are not castable so they trip the warning.